File: srxproperty/src/dataObject/AgentCvPO.js
import { ObjectUtil } from "../utils";
import { AgentAwardPO, TestimonialPO } from "../dataObject";

class AgentCvPO {
  constructor(data) {
    if (!ObjectUtil.isEmpty(data)) {
      this.aboutMe = data.aboutMe;
      this.appointment = data.appointment;
      this.awards = [];
      if (!ObjectUtil.isEmpty(data.awards)) {
        data.awards.map(item => {
          this.awards.push(new AgentAwardPO(item));
        });
      }
      this.credentialAwards = data.credentialAwards;
      this.showListings = data.showListings;
      this.showTransactions = data.showTransactions;
      this.testimonials = [];
      if (!ObjectUtil.isEmpty(data.testimonials)) {
        data.testimonials.map(item => {
          this.testimonials.push(new TestimonialPO(item));
        });
      }
    }
  }
}

export { AgentCvPO };

File: agent-connect-android/app/src/main/java/sg/searchhouse/agentconnect/model/api/agent/AgentCvPO.kt
package sg.searchhouse.agentconnect.model.api.agent

import android.content.Context
import android.text.TextUtils
import com.google.gson.annotations.SerializedName
import sg.searchhouse.agentconnect.util.ApiUtil
import java.io.Serializable

data class AgentCvPO(
    @SerializedName("appointment")
    val appointment: String,
    @SerializedName("aboutMe")
    var aboutMe: String,
    @SerializedName("testimonials")
    val testimonials: ArrayList<Testimonial>,
    @SerializedName("showAboutMe")
    var showAboutMe: Boolean,
    @SerializedName("showListings")
    var showListings: Boolean,
    @SerializedName("showProfile")
    var showProfile: Boolean,
    @SerializedName("showTestimonials")
    var showTestimonials: Boolean,
    @SerializedName("showTransactions")
    var showTransactions: Boolean,
    @SerializedName("publicProfileUrl")
    var publicProfileUrl: String?,
    @SerializedName("changedPublicUrlInd")
    val changedPublicUrlInd: Boolean?
) : Serializable {

    data class Testimonial(
        @SerializedName("clientName")
        var clientName: String,
        @SerializedName("deleted")
        var deleted: Boolean,
        @SerializedName("id")
        val id: Int,
        @SerializedName("photoDeleted")
        val photoDeleted: Boolean,
        @SerializedName("photoUrl")
        val photoUrl: String,
        @SerializedName("tempPhotoId")
        val tempPhotoId: Int,
        @SerializedName("testimonial")
        var testimonial: String,
        @SerializedName("testimonialWriteupLabel")
        val testimonialWriteupLabel: String
    ) : Serializable

    fun getAgentCvUrlWithBaseUrl(context: Context): String {
        publicProfileUrl?.let {
            if (!TextUtils.isEmpty(it)) {
                return "${ApiUtil.getBaseUrl(context)}/${it}"
            }
        }
        return ""
    }
}